src/cascadia/WindowsTerminal/IslandWindow.h

```diff
@@ -26,6 +31,60 @@ class IslandWindow : public BaseWindow<IslandWindow>
 
     void SetCreateCallback(std::function<void(const HWND, const RECT)> pfn) noexcept;
 
+#pragma region IConsoleWindow

```

Comment:
I'm not sure how comfortable I am adding the _entire_ `IConsoleWindow` interface to `IslandWindow`. I was originally hoping to have `IslandWindow` be a general-purpose class for _any_ xaml-islands application, not just us. 
Is there some subset of `IConsoleWindow`'s methods that we need in particular for `WindowUiaProvider`?

Reply:
> I'm not sure how comfortable I am adding the entire IConsoleWindow interface to IslandWindow. I was originally hoping to have IslandWindow be a general-purpose class for any xaml-islands application, not just us.

Is there a reason we should keep it as a general-purpose class in our app? Maybe I can do a bit of refactoring and create a `ConsoleWindow` that extends `IslandWindow` and implements `IConsoleWindow`?

> Is there some subset of IConsoleWindow's methods that we need in particular for WindowUiaProvider?

I'm seeing that we only need `GetWindowHandle()`, `GetWindowRect()`, and `ChangeViewport()` which I made accessible/used in `WindowUiaProviderBase`. The first two are defined in `BaseWindow`, and the third I just have as not implemented right now.

@miniksa mentioned some time ago that some of the other functions might be useful in the future, though. I don't recall the reasoning though.

Reply:
I think there's some discussion about using some of the work we're doing here to help with the interop story in WinUI 3.0. So IslandWindow, NonClientIslandWindow, TitlebarControl, etc. all might end up being parts of WinUI itself. 

`GetWindowHandle()` and `GetWindowRect()` are general enough that they make sense to _all_ `IslandWindow`s, but I don't think that `ChangeViewport` would be. However, if it's some sort of UIA method, then maybe it would make sense? Like other apps might want to be able to implement that method, not just terminal applications.

I think I'd definitely prefer if there was some common `IUiaWindow` interface that `IConsoleWindow` extends. `IslandWindow` would then implement that interface.

In my head for designing this, I'd then have `IslandWindow::ChangeViewport` call a callback that `AppHost` sets, where `AppHost` will then call into the `TerminalApp` to have `TerminalApp` handle the `ChangeViewport` call. (See `IslandWindow::SetCreateCallback` as an example of a similar pattern we're using today.) That way, if someone else were trying to resuse this, they could have their own AppHost (or TerminalApp equivalent) handle the `ChangeViewport` call their own way.

I also probably understand like 15% of the UIA code, so none of that might make any sense.